terraform-modules/aws/istio-networking/main-gateway/main.tf

```diff
@@ -15,6 +15,8 @@ data "template_file" "gateway" {
 
   vars = {
     namespace  = var.namespace
+    gateway_hosts =  "${jsonencode(var.gateway_hosts)}"
```

Comment:
template_file interpolation context doesn't include Terraform's , alternative found it in this github thread : https://github.com/hashicorp/terraform/issues/16775